### PR TITLE
Privatize AC_Avoidance Logging

### PR DESCRIPTION
**Pull request by hendjoshsr71**
This moves avoidance logging functions out of AP_Logger and into AC_Avoidance. Also adds const to where it applies to the logging functions.

Note the `Write_SimpleAvoidance()` function is not called anywhere that I could find....??

**Before PR**
master_OABR
master_OADJ

**After PR**
after_pr_OABR
after_pr_OADJ

### CODE REVIEW

**Comment by rishabsingh3003** [comment]
Hi.
Yes, you are right, I added simple avoidance logging a while back, but then I removed it when we did the avoidance 3D conversion overhaul. I have been meaning to add it back, that's why I didn't remove the supporting functions. Basically, it needs a couple of new fields. I'll do it sometime soon

**Comment by hendjoshsr71** [comment]
> Hi.
> Yes, you are right, I added simple avoidance logging a while back, but then I removed it when we did the avoidance 3D conversion overhaul. I have been meaning to add it back, that's why I didn't remove the supporting functions. Basically, it needs a couple of new fields. I'll do it sometime soon

Ahh makes sense! Figured it might be something like that. Thanks for the clarification

File: libraries/AC_Avoidance/AP_OABendyRuler.cpp [hunk 1]
```diff
@@ -306,7 +306,7 @@ bool AP_OABendyRuler::search_vertical_path(const Location& current_loc, const Lo
                         destination_new.offset_bearing_and_pitch(bearing_to_dest,pitch_delta, distance_to_dest);
                         _current_lookahead = MIN(_lookahead, _current_lookahead * 1.1f);
                     
-                        AP::logger().Write_OABendyRuler((uint8_t)OABendyType::OA_BENDY_VERTICAL, active, bearing_to_dest, pitch_delta, false, margin, destination, destination_new);
+                        Write_OABendyRuler((uint8_t)OABendyType::OA_BENDY_VERTICAL, active, bearing_to_dest, pitch_delta, false, margin, destination, destination_new);
```
Comment (peterbarker) [inline libraries/AC_Avoidance/AP_OABendyRuler.cpp:309]: This shouldn't be casting here, rather casting it in the function.

Pre-existing, so *shrug*.

**Review by peterbarker — APPROVED** [review]
LGTM

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**